libcudacxx/docs/extended_api/synchronization_primitives/barrier/arrive_tx.md

```diff
@@ -0,0 +1,66 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::device::arrive_tx`
+
+Defined in header `<cuda/barrier>`:
+
+```cuda
+__device__
+cuda::barrier<cuda::thread_scope_block>::arrival_token
+cuda::device::arrive_tx(
+  cuda::barrier<cuda::thread_scope_block>& bar,
+  ptrdiff_t arrive_count_update,
+  ptrdiff_t transaction_count_update);
+```
+
+Arrives at a barrier in shared memory, updating both the arrival count and
+transaction count.
+
+## Notes
+
+If `bar` is not in `__shared__` memory, the behavior is undefined. This function
+can only be used under CUDA Compute Capability 9.0 (Hopper) or higher.
+
+To check if `cuda::device::arive_tx` is available, use the
+`__cccl_lib_local_barrier_arrive_tx` feature flag, as shown in the example code below.
+
+## Return Value
+
+An arrival_token, just like the one returned from `cuda::barrier:arrive`.
```

Comment:
```suggestion
An `arrival_token`, just like the one returned from `cuda::barrier::arrive`.
```

Reply:
It now says: `cuda::device::arrive_tx` returns the constructed `arrival_token` object.